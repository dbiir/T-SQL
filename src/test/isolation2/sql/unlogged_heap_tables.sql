-- helpers:
include: helpers/server_helpers.sql;

-- expect: create table succeeds
create unlogged table unlogged_heap_table_managers (
	id int,
	name text
) distributed by (id);

-- skip FTS probes to make the test deterministic.
CREATE EXTENSION IF NOT EXISTS gp_inject_fault;
SELECT gp_inject_fault_infinite('fts_probe', 'skip', 1);
SELECT gp_request_fts_probe_scan();
SELECT gp_request_fts_probe_scan();
SELECT gp_wait_until_triggered_fault('fts_probe', 1, 1);

-- expect: insert/update/select works
insert into unlogged_heap_table_managers values (1, 'Joe');
insert into unlogged_heap_table_managers values (2, 'Jane');
update unlogged_heap_table_managers set name = 'Susan' where id = 2;
select * from unlogged_heap_table_managers order by id;


-- force an unclean stop and recovery:
-- start_ignore
select restart_primary_segments_containing_data_for('unlogged_heap_table_managers');
-- end_ignore

-- expect inserts/updates are truncated after crash recovery
2: select * from unlogged_heap_table_managers;


-- expect: insert/update/select works
3: insert into unlogged_heap_table_managers values (1, 'Joe');
3: insert into unlogged_heap_table_managers values (2, 'Jane');
3: update unlogged_heap_table_managers set name = 'Susan' where id = 2;
3: select * from unlogged_heap_table_managers order by id;

-- force a clean stop and recovery:
-- start_ignore
select clean_restart_primary_segments_containing_data_for('unlogged_heap_table_managers');
-- end_ignore

-- expect: inserts/updates are persisted
4: select * from unlogged_heap_table_managers order by id;

-- expect: drop table succeeds
5: drop table unlogged_heap_table_managers;

SELECT gp_inject_fault('fts_probe', 'reset', 1);
