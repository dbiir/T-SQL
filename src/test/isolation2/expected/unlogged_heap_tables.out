-- helpers:
include: helpers/server_helpers.sql;
CREATE

-- expect: create table succeeds
create unlogged table unlogged_heap_table_managers ( id int, name text ) distributed by (id);
CREATE

-- skip FTS probes to make the test deterministic.
CREATE EXTENSION IF NOT EXISTS gp_inject_fault;
CREATE
SELECT gp_inject_fault_infinite('fts_probe', 'skip', 1);
 gp_inject_fault_infinite 
--------------------------
 Success:                 
(1 row)
SELECT gp_request_fts_probe_scan();
 gp_request_fts_probe_scan 
---------------------------
 t                         
(1 row)
SELECT gp_request_fts_probe_scan();
 gp_request_fts_probe_scan 
---------------------------
 t                         
(1 row)
SELECT gp_wait_until_triggered_fault('fts_probe', 1, 1);
 gp_wait_until_triggered_fault 
-------------------------------
 Success:                      
(1 row)

-- expect: insert/update/select works
insert into unlogged_heap_table_managers values (1, 'Joe');
INSERT 1
insert into unlogged_heap_table_managers values (2, 'Jane');
INSERT 1
update unlogged_heap_table_managers set name = 'Susan' where id = 2;
UPDATE 1
select * from unlogged_heap_table_managers order by id;
 id | name  
----+-------
 1  | Joe   
 2  | Susan 
(2 rows)


-- force an unclean stop and recovery:

-- expect inserts/updates are truncated after crash recovery
2: select * from unlogged_heap_table_managers;
 id | name 
----+------
(0 rows)


-- expect: insert/update/select works
3: insert into unlogged_heap_table_managers values (1, 'Joe');
INSERT 1
3: insert into unlogged_heap_table_managers values (2, 'Jane');
INSERT 1
3: update unlogged_heap_table_managers set name = 'Susan' where id = 2;
UPDATE 1
3: select * from unlogged_heap_table_managers order by id;
 id | name  
----+-------
 1  | Joe   
 2  | Susan 
(2 rows)

-- force a clean stop and recovery:

-- expect: inserts/updates are persisted
4: select * from unlogged_heap_table_managers order by id;
 id | name  
----+-------
 1  | Joe   
 2  | Susan 
(2 rows)

-- expect: drop table succeeds
5: drop table unlogged_heap_table_managers;
DROP

SELECT gp_inject_fault('fts_probe', 'reset', 1);
 gp_inject_fault 
-----------------
 Success:        
(1 row)
